Documentation/flex-to-csi-migration.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Flex Migration
+weight: 11900
+indent: true
+---
+
+# Flex to CSI Migration
+
+In Rook v1.8, the Flex driver has been deprecated. Before updating to v1.8, any Flex volumes created in previous versions of Rook will need to be converted to Ceph-CSI volumes.
+
+The tool [persistent-volume-migrator](https://github.com/ceph/persistent-volume-migrator) will help automate migration of Flex rbd volumes to Ceph-CSI volumes.
+
+## Migration Preparation
+
+1. Rook v1.7.x is required. If you have a previous version of Rook running, follow the [upgrade guide](https://rook.io/docs/rook/v1.7/ceph-upgrade.html) to upgrade from previous releases until on v1.7.x.
+2. Enable the CSI driver if not already enabled. See the [operator settings](https://github.com/rook/rook/blob/release-1.7/cluster/examples/kubernetes/ceph/operator.yaml#L29-L32) such as `ROOK_CSI_ENABLE_RBD`.
+3. Confirm the Rook-Ceph cluster is healthy (`ceph status` shows `health: OK`)
+4. Create the CSI storage class to which you want to migrate
+5. Create the migrator pod
+   1) `kubectl create -f <update this with file link in the tool repo>`
+
+**NOTE**: The Migration procedure will come with a downtime as we need to scale down the applications using the volumes before migration.
+
+## Migrate a PVC
+
+1. Stop the application pods that are consuming the flex volume(s) that need to be converted.
+2. Connect to migration pod
+   1. `migration_pod=$(kubectl -n rook-ceph get pod -l app=rook-ceph-migrator -o jsonpath='{.items[*].metadata.name}')`
+
+   2. `kubectl -n rook-ceph exec -it "$migration_pod" -- sh`
+3. Run the tool to migrate the PVC. See the [section below](#migration-tool-options) for more details on the options and the sample output.
+   1. `<binary_name> --pvc=<pvc-name> --pvc-ns=<pvc-namespace> --destination-sc=<storageclass-name-to-migrate-in> --rook-ns=<rook-ceph> --ceph-cluster-ns=<ceph-cluster-namespace>`
+4. Start the application pods which were stopped in step 1.
+
+
+## Migration Tool Options
+
+These are the options for converting a single PVC. **For more options**, see the [tool documentation](https://github.com/ceph/persistent-volume-migrator), for example to convert all PVCs automatically that belong to the same storage class.
```

Comment:
```suggestion
These are the options for converting a single PVC. For more options, see the [tool documentation](https://github.com/ceph/persistent-volume-migrator), for example to convert all PVCs automatically that belong to the same storage class.
```

Why was this bold?

Reply:
The thought was to get more eyes but yeah we can remove too.